.github/workflows/build-xmtpd.yml

```diff
@@ -14,6 +14,8 @@ jobs:
     permissions:
       contents: read
       packages: write
+    env:
+      DOCKER_METADATA_PR_HEAD_SHA: true
```

Comment:
Noob question - the [doc](https://github.com/docker/metadata-action?tab=readme-ov-file#environment-variables) for this says:
```
If true, set associated head SHA instead of commit SHA that triggered the workflow on pull request event
```
What is the distinction here between the head SHA and the commit SHA?